### PR TITLE
winch: Added monitoring endpoint on different port for sidecar purposes.

### PR DESCRIPTION
**Pull request by Bplotka**
Signed-off-by: Bartek Plotka <bwplotka@gmail.com>

### CODE REVIEW

File: cmd/winch/main.go [hunk 1]
```diff
@@ -154,28 +168,60 @@ func main() {
...
-			if err != nil {
+			if err := srv.Shutdown(ctx); err != nil {
 				log.WithError(err).Errorf("Failed to gracefully shutdown server.")
 			}
 			cancel()
 			httpPlainListener.Close()
 		})
 	}
+	// Setup optional external monitoring address.
+	// This is useful when running winch as a sidecar. Thanks of this we can setup separate endpoint for monitoring.
```
Comment (stefan-improbable) [inline cmd/winch/main.go]: thanks to this instead of of

File: cmd/winch/main.go [hunk 2]
```diff
@@ -154,28 +168,60 @@ func main() {
...
+	// It is not safe to expose normal HTTP and gRPC ports out of the local pod, because there is no auth in front of winch,
+	// so it would allow impersonate attack.
```
Comment (stefan-improbable) [inline cmd/winch/main.go]: delete this line, impersonate attack is obvious

File: cmd/winch/main.go [hunk 3]
```diff
@@ -154,28 +168,60 @@ func main() {
...
+	if *flagMonitoringAddressPort != "" {
+		mux := http.NewServeMux()
+		if *flagMonitoringAddressPort == "" {
```
Comment (stefan-improbable) [inline cmd/winch/main.go]: this is always false

File: cmd/winch/main.go [hunk 4]
```diff
@@ -154,28 +168,60 @@ func main() {
...
-			err := httpWinchServer.Shutdown(ctx)
-			if err != nil {
+			if err := srv.Shutdown(ctx); err != nil {
 				log.WithError(err).Errorf("Failed to gracefully shutdown server.")
 			}
 			cancel()
 			httpPlainListener.Close()
 		})
 	}
+	// Setup optional external monitoring address.
```
Comment (stefan-improbable) [inline cmd/winch/main.go]: can you actually pull this into a separate func, this one is getting a bit long

File: cmd/winch/main.go (hunk 3 above)
Comment (Bplotka) [reply cmd/winch/main.go]: lolol

File: cmd/winch/main.go (hunk 4 above)
Comment (Bplotka) [reply cmd/winch/main.go]: I would vote for leaving it that explicit - main is always long

**Pull request merged**